ws.on is not a function

```
/usr/src/app/node_modules/beepboop-botkit/node_modules/beepboop/lib/resourcer.js:20
      .on('open', handledOpen)
       ^

TypeError: ws.on is not a function
    at Resourcer.resourcer.connect (/usr/src/app/node_modules/beepboop-botkit/node_modules/beepboop/lib/resourcer.js:20:8)
    at new reconnect (/usr/src/app/node_modules/beepboop-botkit/node_modules/beepboop/node_modules/back/reconnect.js:19:12)
    at Back.backoff (/usr/src/app/node_modules/beepboop-botkit/node_modules/beepboop/node_modules/back/index.js:20:20)
    at retry (/usr/src/app/node_modules/beepboop-botkit/node_modules/beepboop/lib/retry.js:18:10)
    at WebSocket.handledError (/usr/src/app/node_modules/beepboop-botkit/node_modules/beepboop/lib/resourcer.js:93:5)
    at emitOne (events.js:77:13)
    at WebSocket.emit (events.js:169:7)
    at ClientRequest.response (/usr/src/app/node_modules/beepboop-botkit/node_modules/beepboop/node_modules/ws/lib/WebSocket.js:696:12)
    at ClientRequest.g (events.js:260:16)
    at emitOne (events.js:77:13)

npm info khaledbot@1.1.0 Failed to exec start script
npm ERR! Linux 3.16.0-4-amd64
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
npm ERR! node v4.2.6
npm ERR! npm  v2.14.12
npm ERR! code ELIFECYCLE
```

--- a/lib/resourcer.js
+++ b/lib/resourcer.js
@@ -14,7 +14,7 @@ module.exports = function Resourcer (config) {
   var resourcer = new EventEmitter2({wildcard: true})
   resourcer.log = log
 
-  resourcer.connect = function (inWs) {
+  resourcer.connect = function () {
     cleanup()
     ws = resourcer.newWebSocket()
     ws

--- a/lib/resourcer.js
+++ b/lib/resourcer.js
@@ -15,7 +15,8 @@ module.exports = function Resourcer (config) {
   resourcer.log = log
 
   resourcer.connect = function (inWs) {
-    ws = initWs(inWs)
+    cleanup()
+    ws = resourcer.newWebSocket()
     ws
       .on('open', handledOpen)
       .on('message', handledMessage)
@@ -37,18 +38,14 @@ module.exports = function Resourcer (config) {
     return cfg
   }
 
-  // cleanup and recreate ws connection for retry and mock case
-  function initWs (inWs) {
+  function cleanup () {
     if (ws) {
       ws.removeAllListeners()
-      // this is a noop if already closed
       ws.close()
     }
-
-    return inWs || newWs()
   }
 
-  function newWs () {
+  resourcer.newWebSocket = function () {
     try {
       var ws = new WebSocket(config.serverURL)
     } catch (e) {

--- a/lib/retry.js
+++ b/lib/retry.js
@@ -2,7 +2,8 @@ var Back = require('back')
 
 var options = {
   minDelay: 1000,
-  maxDelay: 30000
+  maxDelay: 30000,
+  retries: Infinity
 }
 
 var back
@@ -15,6 +16,14 @@ module.exports = function () {
   back = new Back(options)
 
   return function retry (cb) {
-    back.backoff(cb)
+    back.backoff(function (err) {
+      // If backoff fails for some reason, reset it
+      if (err) {
+        back.close()
+        back = new Back(options)
+      }
+
+      cb()
+    })
   }
 }

--- a/test/resourcer_test.js
+++ b/test/resourcer_test.js
@@ -1,5 +1,4 @@
 var events = require('events')
-var WebSocket = require('ws')
 var assert = require('assert')
 var sinon = require('sinon')
 var Resourcer = require('../lib/resourcer.js')
@@ -10,17 +9,18 @@ describe('Resourcer', function () {
   var called
 
   beforeEach(function () {
+    resourcer = new Resourcer()
+
     socket = new events.EventEmitter()
     socket.send = function () {}
-    sinon.stub(WebSocket, 'connect').returns(socket)
+    sinon.stub(resourcer, 'newWebSocket').returns(socket)
 
-    resourcer = new Resourcer()
     resourcer.connect(socket)
     called = false
   })
 
   afterEach(function () {
-    WebSocket.connect.restore()
+    resourcer.newWebSocket.restore()
   })
 
   describe('web socket connection event handling', function () {

